internal/cmds/cmds.go

```diff
@@ -440,41 +449,140 @@ func CopyStateForUpdate(ctx log.Logger, extensionEvents *extensionevents.Extensi
 	return nil
 }
 
-func rehydrateMrSeqFilesForProblematicUpgrades(ctx *log.Context, h types.HandlerEnvironment, extensionEvents *extensionevents.ExtensionEventManager) error {
-	// First, determine whether we're upgrading from a 'problematic' version, defined as one
-	// where we mistakenly deleted the mrseq files in the Disable call
-	newExtensionVersion := os.Getenv(constants.ExtensionVersionEnvName)
-	oldExtensionVersion := os.Getenv(constants.ExtensionVersionUpdatingFromEnvName)
-	newExtensionDirectory := os.Getenv(constants.ExtensionPathEnvName)
-	oldExtensionDirectory := strings.ReplaceAll(newExtensionDirectory, newExtensionVersion, oldExtensionVersion)
-
-	// The following are problematic versions:
-	// Production: 1.3.17
-	// Test: 1.8.0, 1.9.0
-	isProblematicVersion := false
-	isTestExtension := strings.Contains(oldExtensionDirectory, constants.RunCommandTestExtensionName)
-	if isTestExtension {
-		isProblematicVersion = (oldExtensionVersion == constants.FirstTestVersionThatDeletesMrSeqFiles || oldExtensionVersion == constants.SecondTestVersionThatDeletesMrSeqFiles)
-	} else {
-		isProblematicVersion = (oldExtensionVersion == constants.ProductionVersionThatDeletesMrSeqFiles)
+func determineUpgradeVersionDirectories(ctx *log.Context, extensionEvents *extensionevents.ExtensionEventManager) (upgradeFromVersionDirectory string, upgradeToVersionDirectory string, upgradeFromVersion string) {
+	// These two environment variables will tell us the extension versions involved, but won't actually tell us
+	// the from/to versions
+	firstExtensionVersion := os.Getenv(constants.ExtensionVersionEnvName)
+	secondExtensionVersion := os.Getenv(constants.ExtensionVersionUpdatingFromEnvName)
+
+	// To determine to which version we're actually upgrading, we'll need to look into the folders
```

Comment:
For improved correctness, as "disable" is executed first before "update" operation, and update is only executed once, during upgrade of an extension, could we note down the version each time when disable is executed somewhere else where we don't clear or, may be in a file called "/var/lib/waagent/rcv2DisableFromVersion" ? That way we would accurately know what's the version we are updating from. The other version should be updated to  version.
  If this rcv2DisableFromVersion file is not available, we can use mrseq files to determine the from and to versions.